lib/meandro/rules/unused_record_fields.ex

```diff
@@ -31,26 +33,52 @@ defmodule Meandro.Rule.UnusedRecordFields do
     false
   end
 
-  defp analyze_module(file, ast) do
+  defp analyze_module(file, ast, files_and_asts) do
     {_, acc} = Macro.prewalk(ast, %{current_module: nil, records: []}, &collect_record_info/2)
 
-    for {module, name, _num_of_fields, unused_fields, line} <-
+    for {module, name, scope, _num_fields, unused_fields, line} = record <-
           Enum.reverse(acc[:records]),
         unused_field <- unused_fields,
         unused_fields != [] do
       camel_name = name |> Atom.to_string() |> Macro.camelize()
 
-      %Meandro.Rule{
-        file: file,
-        line: line,
-        module: module,
-        text:
-          "Record :#{name} (#{camel_name}) has an unused field in the module: #{unused_field}",
-        pattern: {name, unused_field}
-      }
+      cond do
+        scope == :private ->
```

Comment:
I'd put some comments here, like:
when scope is private, it assumes it's not used in the current module and that's sufficient.